### DISCUSSION TITLE
Enable plugin for certain files only

### DISCUSSION THREAD

**Discussion by xopo** [Exafunction/codeium.nvim#205] [Q&A]
There are some places where I want to get input from Codeium and some others where I don't. AI is a wonderful tool but there are places where I want to be with my own thoughts. How could one set a list of file extensions where the plugin to kick in and make it disabled for all the rest. In my project I don't want .env or .secret to be sent to some remote server and stored or analysed.

**Comment by pqn** [comment]
Soon we will respect a `.codeiumignore` file where you can specify such exclusion/inclusion patterns for all features. Stay tuned.

**Comment by aliaksandr-trush** [comment]
Since codeium.nvim uses nvim-cmp for managing completion, you can configure nvim-cmp plugin to suggest only for some files: https://github.com/hrsh7th/nvim-cmp/issues/1522

**Comment by stefanadelbert** [comment]
I have realised that even if I'm excluding `.env` (which contains sensitive credentials) from my git repo, if I open that file in neovim then those credentials will most likely be sent to Codeium. Or at least I'm not able to control that.

In general, I would like to be able to prevent the content of certain files (buffers) from being sent to Codeium. I would like to be able to do this on filename pattern basis.

@pqn You mentioned upcoming support for `.codeiumignore`, which might fit the bill.

In the meantime, I need a way to do this in neovim. From what I can tell, my LazyVim installation is using `blink.cmp` to trigger Codeium completions. I'm way short of an expert on lua or neovim configuration, but perhaps it's something along these lines in the "right" place,

```lua
    vim.api.nvim_create_autocmd("BufRead", {
      pattern = { "*.env" },
      callback = function()
        -- disable the codeium source for blink.cmp here or something along those lines
      end,
    })
```

I'd really appreciate some guidance on this. I'm very happy for the guidance to be more general too, i.e. prevention of sensitive information leaking from neovim in the new age of external coding agents.
